qutip/lattice.py

```diff
@@ -0,0 +1,814 @@
+
+        if (not isinstance(site, int)):
+            raise Exception("site needs to be int in basis().")
+        elif(site >= self.cell_num_site):
+            raise Exception("site needs to less than Lattice1d.cell_num_site.")
+
+        if(isinstance(dof_ind, int)):
+            dof_ind = [dof_ind]
+
+        if (not isinstance(dof_ind, list)):
+            raise Exception("dof_ind in basis() needs to be an int or \
+                            list of int")
+
+        if (shape(dof_ind) == shape(self.cell_site_dof)):
+            for i in range(len(dof_ind)):
+                if (dof_ind[i] >= self.cell_site_dof[i]):
+                    print("in basis(), dof_ind[", i, "] is required to be \
+                          smaller than cell_num_site[", i, "]")
+                    raise Exception("\n Problem with dof_ind in basis()!")
+        else:
+            raise Exception("dof_ind in basis() needs to be of the same \
+                            dimensions as cell_site_dof.")
+
+        doft = basis(self.cell_site_dof[0], dof_ind[0])
+        for i in range(1, len(dof_ind)):
+            doft = tensor(doft, basis(self.cell_site_dof[i], dof_ind[i]))
+        vec_i = tensor(
+                basis(self.num_cell, cell), basis(self.cell_num_site, site),
+                doft)
+
+        return vec_i
+
+    def distribute_operator(self, op):
+        """
+        A function that returns an operator matrix that applies op to all the
+        cells in the 1d lattice
+
+        Parameters
+        -------
+        op : Qobj
+            Qobj representing the operator to be applied at all cells.
+
+        Returns
+        ----------
+        op_H : Qobj
+            Quantum object representing the operator with op applied at all
+            cells.
+        """
+        nSb = self._H_intra.shape
+        if (not isinstance(op, qutip.qobj.Qobj)):
+            raise Exception("op in distribute_operator() needs to be Qobj.\n")
+        nSi = op.shape
+        if (nSb != nSi):
+            print("\n inter_hop[", i, "] is not dimensionally incorrect.\n")
+            raise Exception("op in distribute_operstor() is required to \
+            have the same dimensionality as cell_Hamiltonian.")
+            cell_All = [i for i in range(self.num_cell)]
+        op_H = self.operator_at_cells(op, cells=cell_All)
+        return op_H
+
+    def x(self):
+        """
+        Returns the position operator. All degrees of freedom has the cell
+        number at their correspondig entry in the position operator.
+
+        Returns
+        -------
+        Qobj(xs) : Qobj
+            The position operator.
+        """
+        nx = self.cell_num_site
+        ne = self._length_for_site
+
+        positions = [(j/ne) for i in range(nx) for j in range(ne)]  # not used
+        # in the current definition of x
+        R = np.kron(range(0, self.num_cell), [1 for i in range(nx * ne)])
+        S = np.kron([1 for i in range(self.num_cell)], positions)
```

Comment:
`[1 for i in range(self.num_cell)]` -> `[1] * self.num_cell` or `np.ones(self.num_cell)`